caffe2/core/storage.h

```diff
@@ -63,8 +103,8 @@ class StorageImpl {
     return capacity_ / itemsize();
   }
 
-  inline void set_device_type(DeviceType device_type) {
-    device_type_ = device_type;
+  void set_numel(int64_t numel) {
+    capacity_ = numel * itemsize();
```

Comment:
Is this necessary?

Reply:
This is added temporarily to remove friending Tensor in Storage. We'll probably move some of the implementations into Storage in the future so we don't need to set capacity_ outside of Storage.

Reply:
OK. Add it as a work item in "nice to do later" and we can keep it.